ODed/BST.cpp

```diff
@@ -44,6 +44,42 @@ struct node *insert(struct node *node, int data)
   return node;
 }
 
+
+
+struct node * minValueNode(struct node* node){
+   struct node* current = node;
+   while (current && current->left != NULL)
+      current = current->left;
+   return current;
+}
+
+struct node* deleteNode(struct node* root, int data){
```

Comment:
not fully functional.

- [ pending ] still need fixes 

- [ pending  ] more readability